packages/py-api/py_api/controllers/hackathon_teams_controller.py

```diff
@@ -0,0 +1,211 @@
+import json
+import random
+import string
+from typing import Any, Dict, List, Optional
+
+from bson.json_util import dumps
+from bson.objectid import ObjectId
+from fastapi.responses import JSONResponse
+from py_api.database.initialize import t_col
+from py_api.models.hackathon_teams_models import (
+    HackathonTeam,
+    MoveTeamMembers,
+    TeamType,
+    UpdateTeam,
+)
+from py_api.utilities.parsers import filter_none_values
+
+
+class TeamsController:
+    @classmethod
+    def create_team(cls, team_name: str, user_id: str = "", members_list: Optional[List[str]] = None) -> (
+            Dict[str, Any] | None
+    ):
+
+        team_type = TeamType.NORMAL
+
+        if not team_name:
+            # If no team_name is provided during registration, the participant is registering individually.
+            # We create a team of type RANDOM which should be filled with such participants
+
+            team_name = ''.join(
+                random.choice(string.ascii_letters)
+                for _ in range(8)
+            )
+            team_type = TeamType.RANDOM
+
+        team = cls.check_if_team_exists(team_name)
+        team_members = []
+
+        if user_id:
+            team_members = [user_id]
+
+        if members_list:
+            team_members = members_list
+            team_type = TeamType.RANDOM
+
+        if not team:
+            new_team = HackathonTeam(
+                team_name=team_name,
+                team_type=team_type, team_members=team_members,
+            )
+            t_col.insert_one(new_team.model_dump())
+
+            return None
+
+        return team
+
+    @classmethod
+    def add_team_participant(cls, team_name: str, user_id: str) -> List[str] | None:
+        team = cls.check_if_team_exists(team_name)
+
+        if not team:
+            return None
+
+        team_members: List[str] = team["team_members"]
+        team_members.append(user_id)
+
+        t_col.update_one(
+            {"team_name": team_name}, {
+                "$set": {"team_members": team_members},
+            }, )
+
+        return team_members
+
+    @classmethod
+    def check_if_team_exists(cls, team_name: str) -> Dict[str, Any] | None:
+        team: Dict[str, Any] = t_col.find_one(filter={"team_name": team_name})
+        return team
+
+    @staticmethod
+    def fetch_teams() -> JSONResponse:
+        teams = list(t_col.find())
+
+        if not teams:
+            return JSONResponse(content={"message": "No teams were found in db"}, status_code=404)
+
+        return JSONResponse(content={"teams": json.loads(dumps(teams))}, status_code=200)
+
+    @staticmethod
+    def get_team(object_id: str) -> JSONResponse:
```

Comment:
Okay, so it seems we have a method which gets a team and also a resolver/controller which has absolutely the same logic.

Now, update the logic of the `check_team_exists` method and invoke it here instead of having repetitive code